code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -549,3 +549,14 @@
 		return TRUE
 	else
 		return FALSE
+
+// ----------------------------
+//  Sci smartfridge
+// ----------------------------
+/obj/machinery/smartfridge/sci
+	desc = "A smart storage vender for tech."
+
+/obj/machinery/smartfridge/sci/accept_check(obj/item/O)
+	if(typecacheof(list(0, /obj/item/stock_parts, /obj/item/disk/tech_disk, /obj/item/circuit_component, /obj/item/assembly)))
```

Comment:
nono no that's not how you do that

you need to make the typecache statically in the `Initialize()` of the smartfridge, then call `is_in_typecache` on the typecache object

Although as a mapper I don't expect you to know how to do this.. so probably just revert it to the old version but remove the trailing slashes.

Reply:
this hurts my brain, but will do!